src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -0,0 +1,296 @@
+            return;
+
+        if (m_eventStarted)
+            return;
+
+        // Get all 4 Gladiators in Range 
+        std::list<Creature*> m_gladiator;
+        GuidVector m_gladiatorGuid;
+        GetCreatureListWithEntryInGrid(m_gladiator, m_creature, NPC_SHATTERED_HAND_GLADIATIOR, 10.0f);
+        for (Creature* gladiator : m_gladiator)
+            m_gladiatorGuid.push_back(gladiator->GetObjectGuid());
+
+        // There should always be 4 Gladiators up handled via spawn_group flag respawn together
+        if (m_gladiatorGuid.size() != 4)
+            return;
+
+        // 2 of them get faction 1693
+        // the other 2 get faction 1692
+        Creature* first = m_creature->GetMap()->GetCreature(m_gladiatorGuid[0]);  // A  
+        Creature* second = m_creature->GetMap()->GetCreature(m_gladiatorGuid[1]); // B
+        Creature* third = m_creature->GetMap()->GetCreature(m_gladiatorGuid[2]); // A
+        Creature* fourth = m_creature->GetMap()->GetCreature(m_gladiatorGuid[3]); // B
+
+        if (first->IsAlive() && second->IsAlive())
+            if (npc_shattered_hand_gladiator* firstGladiator = dynamic_cast<npc_shattered_hand_gladiator*>(first->AI()))
+                firstGladiator->HandleEventStart(second);
+
+        if (third->IsAlive() && fourth->IsAlive())
+            if (npc_shattered_hand_gladiator* thirdGladiator = dynamic_cast<npc_shattered_hand_gladiator*>(third->AI()))
+                thirdGladiator->HandleEventStart(fourth);
+
+        DoBroadcastText(SAY_START, m_creature);
+        m_eventStarted = true;
+        DisableTimer(CENTURION_START_EVENT);
+    }
+
+
+    void HandleRP()
+    {
+        switch (urand(0, 3))
+        {
+            case 0: m_creature->HandleEmote(EMOTE_ONESHOT_LAUGH); break;
+            case 1: m_creature->HandleEmote(EMOTE_ONESHOT_FLEX); break;
+            case 2: m_creature->HandleEmote(EMOTE_ONESHOT_ROAR); break;
+            case 3: m_creature->HandleEmote(EMOTE_ONESHOT_EXCLAMATION); break;
+        }
+    }
+
+    void ReceiveAIEvent(AIEventType eventType, Unit* /*sender*/, Unit* invoker, uint32 /*miscValue*/) override
+    {
+        if (eventType == AI_EVENT_CUSTOM_A)
+        {
+            // Only finish event when Centurion is not in Combat
+            if (m_creature->IsInCombat())
+                return;
+
+            if (m_eventStarted)
+            {
+                m_creature->HandleEmote(EMOTE_ONESHOT_POINT);
+                DoBroadcastText(SAY_FINISH, m_creature);
+
+                CreatureList gladiatorList;
+                GetCreatureListWithEntryInGrid(gladiatorList, m_creature, NPC_SHATTERED_HAND_GLADIATIOR, 20.0f);
+                for (Creature* gladiator : gladiatorList)
+                {
+                    if (gladiator->IsAlive())
+                        SendAIEvent(AI_EVENT_CUSTOM_B, m_creature, gladiator);
+                }
+                m_eventStarted = false;
+                ResetTimer(CENTURION_START_EVENT, urand(10000, 25000));
```

Comment:
I just noticed, this isnt working as i wanted, the timer dont get reset to a rnd timer between 10 and 25 secs, he always trys to start the event after 5 seconds (why 5 not 4??) means the gladiators cant reg to full before fighting again

Reply:
Actually, bug on my end :D

Reply:
Fixed in master